File: backend-ms1/src/main/java/ma/zs/budgetAllocation/service/facade/admin/budget/BudgetCompteBudgetaireAdminService.java
package ma.zs.budgetAllocation.service.facade.admin.budget;

import java.util.List;
import ma.zs.budgetAllocation.bean.core.budget.BudgetCompteBudgetaire;
import ma.zs.budgetAllocation.dao.criteria.core.budget.BudgetCompteBudgetaireCriteria;
import ma.zs.budgetAllocation.zynerator.service.IService;



public interface BudgetCompteBudgetaireAdminService extends  IService<BudgetCompteBudgetaire,BudgetCompteBudgetaireCriteria>  {

    List<BudgetCompteBudgetaire> findByBudgetId(Long id);
    int deleteByBudgetId(Long id);
    long countByBudgetId(Long id);
    List<BudgetCompteBudgetaire> findByCompteBudgetaireId(Long id);
    int deleteByCompteBudgetaireId(Long id);
    long countByCompteBudgetaireId(Long id);




}

File: frontend/src/app/shared/service/admin/budget/BudgetCompteBudgetaireAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {BudgetCompteBudgetaireDto} from 'src/app/shared/model/budget/BudgetCompteBudgetaire.model';
import {BudgetCompteBudgetaireCriteria} from 'src/app/shared/criteria/budget/BudgetCompteBudgetaireCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class BudgetCompteBudgetaireAdminService extends AbstractService<BudgetCompteBudgetaireDto, BudgetCompteBudgetaireCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): BudgetCompteBudgetaireDto {
        return new BudgetCompteBudgetaireDto();
    }

    public constrcutCriteria(): BudgetCompteBudgetaireCriteria {
        return new BudgetCompteBudgetaireCriteria();
    }

    get API() {
        return environment.apiUrlBudgetAllocationservice + 'admin/budgetCompteBudgetaire/';
    }
}
